### DISCUSSION TITLE
[PorousFlow] Volume fraction and BCs

### DISCUSSION THREAD

**Discussion by 1runer** [idaholab/moose#24001] [Q&A General] [answered]
Hi,

I currently use PorousFlow with volume flow using GlobalParams multiply_by_density = false.
My problem is 1 phase - 2 component flow, pore pressure and a tracer.

I want to inject a constant volume flow of a mixture at an inlet, so for example 1 m3/s with volume fraction component_0 = 0.1 (tracer) and component_1 = 0.9 (pore pressure), and I'm not sure which BCs to apply and which 'use_' options to put there. 
Unlike most kernels, PorousFlowSink doesn't provide the 'multiply_by_density = false' option and I'm not sure if it's even needed there. So the explanation on the homepage is not quite clear to me. 

```
[GlobalParams]
  PorousFlowDictator = dictator
  gravity = '0 0 -9.81'
  multiply_by_density = false        # make it volumetric
[]
[BCs]
  [pout]
    type = DirichletBC # ensure outlet on model top 
    variable = pp
    boundary = outlet_area
    value = 0
  []
  [flow_inj]
    type = PorousFlowSink  # constant tracer injection; vol. flow rate = 1 m3/s with 0.1 tracer vol concentration
    boundary = 'inlet_area'
    variable = pp
    flux_function = '-0.9' # inlet area is 1x1 m, so q(inj)= 1m/s
    use_relperm = true 
    fluid_phase = 0
 []
 [comp_inj]
    type = PorousFlowSink
    boundary = 'inlet_area'
    variable = volfrac0
    flux_function = '-0.1'
    use_relperm = true
    fluid_phase = 0
  []
[]
```

I am unsure if I am even on the right track with this approach?
Best regards

**Comment by GiudGiud** [comment]
Hello

The porousFlowSink contributes : function * test * local relative permeability.

So as long as the function is in the same units as your variables, and you are consistent about using the relative permeability, I think you are on the right track.

@WilkAndy the function parameter description says the unit is `kg.m^-2.s^-1`. Is this still true when in 'volume flow' with `multiply_by_density = false` everywhere else?

Guillaume

**Comment by WilkAndy** [comment, marked as answer]
Great question @1runer .    When you use `multiply_by_density = false`, the `flux_function` of [PorousFlowSink](https://mooseframework.inl.gov/modules/porous_flow/boundaries.html) is measured in m^3.m^-2.s^-1.

Please think carefully about whether you want to use `multiply_by_density = false`.  Eg, quoting from [PorousFlowFullySaturated](https://mooseframework.inl.gov/source/actions/PorousFlowFullySaturated.html):
"When not multiply by density, in many situations the sum of [Eq. (1)](https://mooseframework.inl.gov/source/actions/PorousFlowFullySaturated.html#moose-equation-9e7a0124-8cb4-4820-89f7-673bdafc1cac) will imply a Laplace equation such 0=∇(k(∇P−ρg)), that is a steady-state distribution for the porepressure. In addition, care must be taken when using other parts of PorousFlow, for instance, the [PorousFlowMass](https://mooseframework.inl.gov/source/postprocessors/PorousFlowFluidMass.html) Postprocessor is coded to record fluid mass, not fluid volume"

**Reply by 1runer** [reply]
Thanks @WilkAndy  and @GiudGiud  for the confirmation. 

The idea is that it is rather common in the literature and also in other codes to calculate component transport and saturation in volume fractions and accordingly most material properties or mixing equations are designed for volume flow. I am aware that for the dissolution of salts and gases it is somewhat different considering molar masses etc.  

"Eq. (1) PorousFlowFullySaturated will imply a Laplace equation such 0=∇(k(∇P−ρg))"
My very naive (and probably stupid) question: what are the practical implications for the results or simulation of multiplying both sides of the equation by 1 rather than the fluid density? For convergence reasons this is understandable. Is there further literature regarding this?
